### DISCUSSION TITLE
Runtime locale configuration

### DISCUSSION THREAD

**Discussion by jonlambert** [vercel/next.js#42691] [Help]
Hey everyone.

Currently, it's possible to configure a set of supported locales in `next.config.js`. 

```
{
  i18n: {
    locales: ['en', 'fr', 'es', 'de'],
    defaultLocale: 'en',
    localeDetection: true,
  }
}
```

I'm currently working on a multi-tenant application build with Next.js that has a different list of locales fetched at runtime.

Is it possible to update this configuration ^ at runtime? Or do we need to plan to circumvent Next.js' locale redirection / routing support and build our own with middleware?

Thanks in advance for any suggestions!

**Comment by 87Hz** [comment]
@jonlambert, have you found solution already? We got it working for our multi-tenant use case by using [custom server](https://nextjs.org/docs/pages/building-your-application/configuring/custom-server).
Thanks.
